ibis/backends/tests/base.py

```diff
@@ -123,7 +123,7 @@ def least(
         return f(*args)
 
     @property
-    def db(self) -> ibis.client.Database:
+    def db(self):
```

Comment:
why are you removing typing?

Reply:
I updated the location of `Database`, and having annotations in tests doesn't seem so useful, so just deleted them.

I can restore them to the right location if you think they are important. But we don't have so many annotations in the code, and having those tests annotated is annoying when refactoring, so to meet they cause more harm than good. But not a big deal.

This PR is not very big, it's just that lots of imports had to be updated. Btw, #2809 is ready, and it's more useful to me to have it merge sooner, since it's blocking other refactorings of the compiler).